### PR TITLE
added tests for and refactored classroom manager controller

### PR DESCRIPTION
**Pull request by gauravrakheja**
Addresses issue #

**Changes proposed in this pull request:**
- added test for classroom manager controller
- refactored classroom manager controller

**Has this branch been QA'd on staging?** no

**Have you updated the docs?** no

**Have the tests been updated?** yes

**Reviewer:** @ddmck

### CODE REVIEW

File: app/controllers/teachers/classroom_manager_controller.rb [hunk 1]
```diff
@@ -219,6 +181,53 @@ def import_google_students
 
   private
 
+  def set_classroom_variables
+    @tab = params[:tab] #|| "manageUnits"
```
Comment (ddmck) [inline app/controllers/teachers/classroom_manager_controller.rb]: lets get rid of the unreachable code in the comment.

File: app/controllers/teachers/classroom_manager_controller.rb [hunk 2]
```diff
@@ -39,21 +31,11 @@ def generic_add_students
   end
 
   def retrieve_classrooms_for_assigning_activities # in response to ajax request
-    classrooms_and_their_students = current_user.classrooms_own.map do |classroom|
-      {  classroom: classroom, students: classroom.students.sort_by(&:sorting_name)}
-    end
-    render json: {
-      classrooms_and_their_students: classrooms_and_their_students
-    }
+    render json: classroom_with_students_json(current_user.classroom_own)
```
Comment (ddmck) [inline app/controllers/teachers/classroom_manager_controller.rb]: Do we need to change this to classrooms_i_own? And does this mean we can un-comment the test that fails due to this error?

File: spec/controllers/teachers/classroom_manager_controller_spec.rb [hunk 3]
```diff
@@ -35,4 +39,422 @@
...
+
+      it 'should assign the classroom and redirect to profile path' do
+        get :generic_add_students
+        expect(assigns(:classroom)).to eq teacher.classrooms_i_teach.first
+        expect(response).to redirect_to invite_students_teachers_classrooms_path
+      end
+    end
+  end
+
+  # this controller method calls current_user.classroom_own which is not defined
```
Comment (ddmck) [inline spec/controllers/teachers/classroom_manager_controller_spec.rb]: See other comment about updating function to use classrooms_i_own

File: spec/controllers/teachers/classroom_manager_controller_spec.rb [hunk 4]
```diff
@@ -35,4 +39,422 @@
...
+      end
+
+      it 'should set the firewall test to true' do
+        get :dashboard
+        expect(assigns(:firewall_test)).to eq true
+      end
+    end
+  end
+
+  # describe '#students_list' do
```
Comment (ddmck) [inline spec/controllers/teachers/classroom_manager_controller_spec.rb]: Why exactly is this test failing? Can you paste the error message?

File: spec/controllers/teachers/classroom_manager_controller_spec.rb [hunk 5]
```diff
@@ -35,4 +39,422 @@
...
+        end
+      end
+    end
+  end
+
+  describe '#generic_add_students' do
+    let(:student) { create(:student) }
+    let(:teacher) { create(:teacher) }
+
+    # the before filter authorize_teacher! redirects the user so this branch is never traversed
```
Comment (ddmck) [inline spec/controllers/teachers/classroom_manager_controller_spec.rb]: Can we write a test that confirms the controller action the routes is never called then?

File: app/controllers/teachers/classroom_manager_controller.rb (hunk 2 above)
Comment (gauravrakheja) [reply app/controllers/teachers/classroom_manager_controller.rb]: Yes changing this to classrooms I own should make the test pass

File: app/controllers/teachers/classroom_manager_controller.rb (hunk 1 above)
Comment (gauravrakheja) [reply app/controllers/teachers/classroom_manager_controller.rb]: Sure

File: spec/controllers/teachers/classroom_manager_controller_spec.rb (hunk 4 above)
Comment (gauravrakheja) [reply spec/controllers/teachers/classroom_manager_controller_spec.rb]: Its passing in local environment but for some reason the classroom is not found on Travis county and it gives a undefinef method students for nil error

File: spec/controllers/teachers/classroom_manager_controller_spec.rb (hunk 5 above)
Comment (gauravrakheja) [reply spec/controllers/teachers/classroom_manager_controller_spec.rb]: do you mean adding a test for the before filter?

Comment (ddmck) [reply spec/controllers/teachers/classroom_manager_controller_spec.rb]: If possible, yes

**Pull request merged**